File: CLASE II - TDA/clase_2.2_ayed/clase_1_ayed/c/persona.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "funciones.h"
#include "persona.h"


struct PersonaE{ //.nombre

    int dni;
    char nombre[15];

    //Un solo carton
    CartonPtr cartonDeLaPersona;

};

int getDNI(PersonaPtr p){
    return p->dni;

};
char* getNombre(PersonaPtr p){

    return p->nombre;
};

CartonPtr getCartonPtr(PersonaPtr p){

    return p->cartonDeLaPersona;
};

void setDNI(PersonaPtr p, int nuevoDNI){

    p->dni = nuevoDNI;
};

void setNombre(PersonaPtr p, char nuevoNombre[15]){

    strcpy(p->nombre, nuevoNombre);
};

void setCartonPtr(PersonaPtr p, CartonPtr nuevoCarton){

    p->cartonDeLaPersona = nuevoCarton;

};


//Constructores
PersonaPtr crearPersonaTeclado(); //Teclado

PersonaPtr crearPersona(int d, char n[15]){

    PersonaPtr p = (PersonaPtr) malloc(sizeof(struct PersonaE)); //void *

    p->dni = d;
    strcpy(p->nombre, n);

    p->cartonDeLaPersona = crearCartonAutomatico("Vacio");

    return p;


};

PersonaPtr crearPersonaYCarton(int d, char n[15], char dir[15]){

    PersonaPtr p = crearPersona(d,n);

   //strcpy( p->cartonDeLaPersona->direccion, dir); //Cambio para TDA

   setDireccion(p->cartonDeLaPersona, dir);


    return p;
}

;


//Destructor
void destruirPersona(PersonaPtr p){

    free(p);


};

//Mostrar
void mostrarPersona(PersonaPtr p){


    printf("\n---------- PERSONA -----------\n");
    printf("DNI: %d\n", p->dni);
    printf("NOMBRE: %s\n", p->nombre);

    mostrarCarton(p->cartonDeLaPersona);

};

File: CLASE II - TDA/clase_2.2_ayed/clase_1_ayed/golang/persona.go
package main

import "fmt"

type Persona struct {
	dni    int
	nombre string
	carton *Carton
}

func (p *Persona) GetDNI() int {
	return p.dni
}

func (p *Persona) GetNombre() string {
	return p.nombre
}

func (p *Persona) GetCartonPtr() *Carton {
	return p.carton
}

func (p *Persona) SetDNI(nuevoDNI int) {
	p.dni = nuevoDNI
}

func (p *Persona) SetNombre(nuevoNombre string) {
	p.nombre = nuevoNombre
}

func (p *Persona) SetCartonPtr(nuevoCarton *Carton) {
	p.carton = nuevoCarton
}

func crearPersona(dni int, nombre string) *Persona {
	return &Persona{
		dni:    dni,
		nombre: nombre,
		carton: crearCartonAutomatico("Vacio"),
	}
}

func crearPersonaYCarton(dni int, nombre string, dir string) *Persona {
	p := crearPersona(dni, nombre)
	p.carton.SetDireccion(dir)
	return p
}

func mostrarPersona(p *Persona) {
	fmt.Printf("\n---------- PERSONA -----------\n")
	fmt.Printf("DNI: %d\n", p.dni)
	fmt.Printf("NOMBRE: %s\n", p.nombre)
	mostrarCarton(p.carton)
}

